infrastructure/src/test/java/org/corfudb/infrastructure/ServerHandshakeHandlerTest.java

```diff
@@ -0,0 +1,183 @@
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.corfudb.protocols.service.CorfuProtocolMessage.ClusterIdCheck;
+import org.corfudb.protocols.service.CorfuProtocolMessage.EpochCheck;
+import org.corfudb.runtime.proto.service.CorfuMessage.HeaderMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.PriorityLevel;
+import org.corfudb.runtime.proto.service.CorfuMessage.ProtocolVersionMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.RequestMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.ResponseMsg;
+import org.corfudb.util.GitRepositoryState;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.corfudb.protocols.CorfuProtocolCommon.DEFAULT_UUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolBase.getHandshakeRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolBase.getPingRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getRequestMsg;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ServerHandshakeHandlerTest {
+
+    // Netty EmbeddedChannel for unit testing.
+    private EmbeddedChannel embeddedChannel;
+
+    // Some utility variables needed for following test cases.
+    private final AtomicInteger requestCounter = new AtomicInteger();
+
+    private static final String DEFAULT_HANDSHAKE_TIMEOUT = "10";
+
+    private static final long FAKE_CLIENT_VERSION = 111111111;
+
+    private static final UUID SERVER_NODEID = UUID.randomUUID();
+
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId   indicates if the message is clusterId aware
+     * @param ignoreEpoch       indicates if the message is epoch aware
+     * @return                  the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(ClusterIdCheck ignoreClusterId, EpochCheck ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    @Before
+    public void setup() {
+        ServerHandshakeHandler serverHandshakeHandler = new ServerHandshakeHandler(SERVER_NODEID,
+                GitRepositoryState.getCorfuSourceCodeVersion(), DEFAULT_HANDSHAKE_TIMEOUT);
```

Comment:
Since this is a unit test, maybe the server version can be mocked as well so it doesn't have to rely on GitRepositoryState.

Reply:
Changed, thanks!